third_party/dependency_analyzer/src/main/io/bazel/rulesscala/dependencyanalyzer3/DependencyAnalyzerSettings.scala

```diff
@@ -0,0 +1,121 @@
+package io.bazel.rulesscala.dependencyanalyzer
+
+object AnalyzerMode {
+  case object Error extends AnalyzerMode
+  case object Warn extends AnalyzerMode
+  case object Off extends AnalyzerMode
+
+  def parse(mode: String): Option[AnalyzerMode] = {
+    mode match {
+      case "error" => Some(Error)
+      case "warn" => Some(Warn)
+      case "off" => Some(Off)
+      case _ => None
+    }
+  }
+}
+
+sealed trait AnalyzerMode
+
+object DependencyTrackingMethod {
+  case object HighLevel extends DependencyTrackingMethod("high-level")
+
+  /**
+   * Discovers dependencies by crawling the AST.
+   */
+  case object Ast extends DependencyTrackingMethod("ast")
+
+  def parse(mode: String): Option[DependencyTrackingMethod] = {
+    Seq(HighLevel, Ast).find(_.name == mode)
+  }
+}
+
+sealed abstract class DependencyTrackingMethod(val name: String)
+
+class TargetSet(
+  prefix: String,
+  jarsSeq: Seq[String],
+  targetsSeq: Seq[String]
+) {
+  private lazy val jarToTargetMap: Map[String, String] = {
+    require(targetsSeq.size == jarsSeq.size, s"Arguments $prefix-jars and $prefix-targets had mismatched size")
+
+    jarsSeq.zip(targetsSeq).toMap
+  }
+
+  def targetFromJarOpt(jar: String): Option[String] = {
+    jarToTargetMap.get(jar)
+  }
+
+  lazy val jarSet: Set[String] = {
+    jarsSeq.toSet
+  }
+}
+
+object DependencyAnalyzerSettings {
+  def parseSettings(
+    options: List[String],
+    error: String => Unit
+  ): DependencyAnalyzerSettings = {
+
+    val optionsParser = OptionsParser.create(options, error)
+
+    def decodeTarget(target: String): String = {
+      target.replace(";", ":")
+    }
+
+    def parseTargetSet(prefix: String): TargetSet = {
+      new TargetSet(
+        prefix = prefix,
+        jarsSeq = optionsParser.takeStringSeqOpt(s"$prefix-jars").getOrElse(Seq.empty),
+        targetsSeq = optionsParser.takeStringSeqOpt(s"$prefix-targets").map(_.map(decodeTarget)).getOrElse(Seq.empty)
+      )
+    }
+
+    def extractAnalyzerMode(key: String): AnalyzerMode = {
+      optionsParser
```

Comment:
Default are a bit confusing. If unsupported value is passed then it default to `error` if no value defaults to `off`. Maybe default should be off all other values should fail.